ITE/10/README.adoc

```diff
@@ -0,0 +1,288 @@
+is that inspection scripts must also reason about and verify any prior metadata
+they're looking at is signed by the right keys, with the right threshold, and so
+on. They cannot just verify the claims themselves. This is important because:
+1. you may have link metadata that was signed by the wrong functionary, rejected
+   during verification of the step, but the threshold was still met. Claims made
+   by this metadata shouldn't be trusted by inspections.
+2. you may have link metadata with differing attributes for the same step by
+   design. One example: reproducible builds with one piece of metadata generated
+   by an isolated environment. The inspection here should be able to reason
+   about which metadata has these claims and if it's from the right environment.
+
+Note that a policy specified in an inspection block is not applied to some
+specified set of attestation types. Policies apply to _subjects_ of the supply
+chain, whether artifacts or other processes captured using ITE-4 semantics.
+During the verification workflow, all attestations are parsed to represent their
+claims in the context of the subjects they apply to. So, all available
+information for a subject is available along with the functionaries signing for
+each individual claim, and this representation of attestation claims is used to
+apply policies. Attestations must not be parsed if their signatures cannot be
+verified using functionary information available in the layout.
+
+=== Specifying Policies
+
+In general, each unit policy applies one constraint. For example, a policy might
+mirror in-toto's artifact rules, ensuring that all the materials and products of
+some step in the supply chain are as expected. Alternatively, a policy can apply
+a property-oriented constraint to an artifact or step, such as validating the
+Provenance of an artifact or that its build process was performed in an isolated
+and hermetic build environment. Note that all of these types of policies can
+coexist and successful verification is contingent on all policy validations
+succeeding.
+
+This indicates that developers need a great deal of flexibility in specifying
+policies _and_ that any domain-specific language chosen or created to express
+policies must have the necessary capabilities to fit these requirements.
+
+Prior literature has explored the use of logic based languages and their
+derivatives for specifying and validating policies. Policies, at their core, are
+logic programs that are applied against a set of claims, expressed here as
+in-toto attestations. This ITE uses languages such as Rego and CUE as examples
+of such languages, but leaves the specific choices to the in-toto community and
+adopters. The new layout schema proposed above includes a `<POLICY_LANGUAGE>`
+attribute for each policy to enable this variety in choice.
+
+=== Validating Policies
+
+The in-toto verification workflow is updated to support different types of
```

Comment:
How should a verifier call out to these policy languages? Seems completely unspecified to me, especially error handling.

Reply:
I think underspecifying is bad too. How will the attestations be made available to the policy environment? That needs to be documented. If the ITE doesn't do that, we should at least say where it will happen